### ISSUE TITLE
Ensure `active_hkls` indices are used on filtered hkls

### ISSUE THREAD

**Issue by psavery** [HEXRD/hexrd#188]
@joelvbernier I'm just making a note here so we don't forget it.

"filtered hkls" here means hkls where exclusions and tThMax were applied to remove hkls.

The `active_hkls` indices are used to index into a list of filtered hkls. But there is at least one place
where it looks like `active_hkls` are indexing into a list of unfiltered hkls:

https://github.com/HEXRD/hexrd/blob/e9b01fa682ae5b5fefb4c57ba349a4db37385adf/hexrd/findorientations.py#L589

We should make sure active_hkl indices are only used to index into filtered hkls.

**Comment by joelvbernier** [comment]
setting `PlaneData.tThMax` does modify exclusions.

**Comment by joelvbernier** [comment]
@psavery -- I just wanted to follow up on this.  Did we resolve this?

**Comment by psavery** [comment]
I don't think so. Basically, there is a problem only if some hkls are skipped (other than trailing hkls). Here's an example. For the [single GE example](https://github.com/HEXRD/examples/blob/master/NIST_ruby/single_GE/include/ruby_config.yml), since it has a min structure factor of 0.05, the hkls at indices 3 and 5 and  gets skipped.

If you change the [active hkls to just [5]](https://github.com/HEXRD/examples/blob/77671d621f42d3852d44eab24e36f78ba47a3a24/NIST_ruby/single_GE/include/ruby_config.yml#L39), and the [hkl seeds to just [0]](https://github.com/HEXRD/examples/blob/77671d621f42d3852d44eab24e36f78ba47a3a24/NIST_ruby/single_GE/include/ruby_config.yml#L48), and you add this print statement to [this line](https://github.com/HEXRD/hexrd/blob/16be327084c422acb6ac0c03a416949874aa58f5/hexrd/findorientations.py#L657):
```python
    hkl_strings = [
        plane_data.hklDataList[active_hkls[i]]['hkl']
        for i in fiber_seeds
    ]
    print('In create_clustering_parameters, the hkls being used are:',
          hkl_strings)
```

You will see from the log output:
```
21-03-24 13:09:43 - building eta_ome maps using hkls: [1 1 6]
...
In create_clustering_parameters, the hkls being used are: [array([2, 0, 2], dtype=int32)]
```

Basically, the hkls don't match. This is because the the `plane_data.hklDataList` doesn't filter out the excluded hkls, but everywhere else that `active_hkls` is used is being used, the excluded hkls do get filtered out.

**Comment by joelvbernier** [comment]
I'll look at this one tomorrow @psavery

**Comment by joelvbernier** [comment]
I think I have a solution for this; just needs testing.

**Comment by joelvbernier** [comment]
@psavery: is this still an issue?

**Comment by psavery** [comment]
Yeah, I don't think the code was fixed
